test/extend_mongodb_tests.js

```diff
@@ -1,218 +1,161 @@
-        // Create a doc
-        var doc1 = {
-          int32: new Int32(10)
-        };
-
-        // Serialize the document
-        var text = extJSON.stringify(doc1, null, 0);
-        expect(text).to.equal('{"int32":{"$numberInt":"10"}}');
-
-        // Deserialize the json in strict and non strict mode
-        var doc2 = extJSON.parse(text, { strict: true });
-        expect(doc2.int32._bsontype).to.equal('Int32');
-        doc2 = extJSON.parse(text, { strict: false });
-        expect(doc2.int32).to.equal(10);
-      });
-
-      it('should correctly serialize bson types when they are values', function() {
-        var Int32 = extJSON.BSON.Int32;
-        var serialized = extJSON.stringify(new ObjectID('591801a468f9e7024b6235ea'));
-        expect(serialized).to.equal('{"$oid":"591801a468f9e7024b6235ea"}');
-        serialized = extJSON.stringify(new Int32(42));
-        expect(serialized).to.equal('{"$numberInt":"42"}');
-        serialized = extJSON.stringify({
-          _id: { $nin: [new ObjectID('591801a468f9e7024b6235ea')] }
-        });
-        expect(serialized).to.equal('{"_id":{"$nin":[{"$oid":"591801a468f9e7024b6235ea"}]}}');
-
-        serialized = extJSON.stringify(new Binary(new Uint8Array([1, 2, 3, 4, 5])));
-        expect(serialized).to.equal('{"$binary":{"base64":"AQIDBAU=","subType":"00"}}');
-      });
-
-      it('should correctly parse null values', function() {
-        expect(extJSON.parse('null')).to.be.null;
-        expect(extJSON.parse('[null]')[0]).to.be.null;
-
-        var input = '{"result":[{"_id":{"$oid":"591801a468f9e7024b623939"},"emptyField":null}]}';
-        var parsed = extJSON.parse(input, { strict: false });
-
-        assert.deepEqual(parsed, {
-          result: [{ _id: new ObjectID('591801a468f9e7024b623939'), emptyField: null }]
-        });
-      });
-
-      it('should correctly throw when passed a non-string to parse', function() {
-        expect(() => {
-          extJSON.parse({}, { strict: true });
-        }).to.throw;
-      });
-
-      it('should allow relaxed parsing', function() {
-        const dt = new Date(1452124800000);
-        const inputObject = {
-          int: { $numberInt: '500' },
-          long: { $numberLong: '42' },
-          double: { $numberDouble: '24' },
-          date: { $date: { $numberLong: '1452124800000' } }
-        };
-
-        const parsed = extJSON.parse(JSON.stringify(inputObject), { relaxed: true });
-        expect(parsed).to.eql({
-          int: 500,
-          long: 42,
-          double: 24,
-          date: dt
-        });
-      });
-
-      it('should allow regexp', function() {
-        const parsedRegExp = extJSON.stringify({ test: /some-regex/i }, { relaxed: true });
-        const parsedBSONRegExp = extJSON.stringify(
-          { test: new BSONRegExp('some-regex', 'i') },
-          { relaxed: true }
-        );
-        expect(parsedRegExp).to.eql(parsedBSONRegExp);
-      });
+    expect(parsed).to.deep.equal({
```

Comment:
@daprahamian @mbroadst @kvwalker This test was failing when using the assert.deepEqual method, but succeeded using Chai's deep equal method. I experimented and found the following: 
```
const test1 = { 'a': 1 }; 
const test2 = { a: 1 };

expect(test1).to.deep.equal(test2); // succeeds
assert.deepEqual(test1, test2); // fails
```
The difference is just the quotes around object keys. In this case, Is it acceptable that assert.deepEqual fails while the Chai method succeeds?

Reply:
Just kidding, the problem was something else - error msg: 
```
✖ should correctly parse null values
      HeadlessChrome 0.0.0 (Mac OS X 10.13.3)
    TypeError: buffer_1 is not a function
        at _deepEqual (test/extend_mongodb_tests.js:4833:14)
        at Function.deepEqual (test/extend_mongodb_tests.js:4818:8)
        at Context.<anonymous> (test/extend_mongodb_tests.js:23658:14)
```

Reply:
Based on that error, it looks like this is a polyfill / rollup issue, and not an actual assertion failure. I'd replace it with chai.

Reply:
it seems like `extJSON.parse` above should be wrapped in a `try/catch`?

Reply:
@mbroadst since this is a synchronous test, a failure on the parse will still cause the test to fail.